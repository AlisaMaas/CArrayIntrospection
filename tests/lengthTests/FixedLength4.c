/**
* This tests the range analysis results when there are conditionals in place.
* May be used at some future point when we have support for predicated types.
**/
int foo(int* array, int x){
	int index = 0;
	if (x > 3) {
		index = 8;
	}
	else {
		index = 2; 
	}
	int sum = 0;
	for (int i = 0; i < index; i++) {
		sum += array[i];
	}
	return sum;
	
}