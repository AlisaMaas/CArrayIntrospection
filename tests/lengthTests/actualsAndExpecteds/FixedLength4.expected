Analyzing foo
	Argument array has max index 7
	Argument x has unknown max index.
